pkg/binary/envoy/controlplane/istio.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/tetratelabs/getenvoy/pkg/binary"
+	"github.com/tetratelabs/getenvoy/pkg/binary/envoy"
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/pkg/log"
+
+	"istio.io/istio/pilot/pkg/model"
+	agent "istio.io/istio/pkg/bootstrap"
+	"istio.io/istio/pkg/config/mesh"
+)
+
+const (
+	defaultControlplane   = "istio-pilot:15010"
+	initialEpochBootstrap = "envoy-rev1.json"
+)
+
+// Istio tells GetEnvoy that it's using Istio for xDS and should bootstrap accordingly
+var Istio = func(r *envoy.Runtime) {
```

Comment:
I'd rather see this be `func Istio(r *envoy.Runtime)` - at your usage site it's identical, but that way you can't re-assign this function.